### PR TITLE
feat: add "related chats" to group profile

### PR DESCRIPTION
**Pull request by link2xt**
This is using experimental core API https://github.com/deltachat/deltachat-core-rust/pull/4649

For merging:
- [x] Adapt to new API once the core is updated https://github.com/deltachat/deltachat-core-rust/pull/4693
- [x] Rename from "Shared Chats" into "Related Chats" in the group profile and add a translation string
- [x] Hide by default and add a new experimental setting to enable it

### CODE REVIEW

**Comment by ralphtheninja** [comment]
> I have rebased it now, but it needs latest stable core v1.121.0 instead of v1.120.0 that is currently in the `package.json` file.

So it can be rebased again on top of `v1.40.4`?

**Comment by ralphtheninja** [comment]
Maybe name it "Related Chats" instead of "Shared Chats" in the ui?

**Comment by hpk42** [comment]
does it need  a further rebase?  For me this branch crashes but i think it's something that was on master but was fixed now?

**Comment by ralphtheninja** [comment]
> does it need a further rebase? For me this branch crashes but i think it's something that was on master but was fixed now?

No further rebase is needed. Works fine on my machine (clean `npm install`, `npm run build` and `npx electron .`).  Still crashing on yours?

**Comment by hpk42** [comment]
i deleted my local branch, then checked it out again, rmoved node-modules, rebuild desktop (with latest core stable, also made sure to have the latest) but i get a crash when click the group profile.  
image

**Comment by hpk42** [comment]
On Mon, Sep 18, 2023 at 03:14 -0700, link2xt wrote:
> There has been a change to the API on https://github.com/deltachat/deltachat-core-rust/pull/4693
> PR works with core 1.121.0 as specified in package.json, but needs to be adapted for newer core.

that explains it.  I am typically building from core/stable also to test out 
the most recent desktop/core functioning -- but will now try v1.121.0 to get related chats back.

File: src/renderer/components/dialogs/ViewGroup.tsx [hunk 1]
```diff
@@ -238,6 +265,32 @@ function ViewGroupInner(props: {
...
+                <ChatListPart
+                  isRowLoaded={isChatLoaded}
+                  loadMoreRows={loadChats}
+                  rowCount={chatListIds.length}
+                  width={400}
+                  height={CHATLISTITEM_CHAT_HEIGHT * chatListIds.length}
+                  itemKey={index => 'key' + chatListIds[index]}
+                  itemHeight={CHATLISTITEM_CHAT_HEIGHT}
+                >
+                  {({ index, style }) => {
```
Comment (ralphtheninja) [inline src/renderer/components/dialogs/ViewGroup.tsx:281]: React newb question: How does passing functions to components work in react?

Comment (ralphtheninja) [reply src/renderer/components/dialogs/ViewGroup.tsx:281]: Nvm. So this function here gets passed as `children` in `ChatListPart` which then passes that on to `FixedSizeList` https://web.dev/virtualize-long-lists-react-window/

**Review by missytake — APPROVED** [review]
Didn't try around very much and didn't try to break it, but it runs on xubuntu 22.04 and looks and feels nice!

**Pull request merged**